tools/plugins/com.liferay.ide.upgrade.plan.ui/src/com/liferay/ide/upgrade/plan/ui/internal/UpgradePlanView.java

```diff
@@ -161,6 +172,95 @@ public void setSelection(ISelection selection) {
 		_upgradePlanElementViewer.setSelection(selection);
 	}
 
+	private void _changeSelection(ISelection selection, boolean deepFind, boolean findFirstLeaf) {
```

Comment:
@simonjhy   Please move this logic and behavior all inside of the UpgradePlanViewer instead of in this class.  That way the `UpgradePlanView` does not need to keep references to the `_treeViewer` and `_treeContentProvider`.  Just move all of the listener logic inside of the `UpgradePlanViewer`.  Thanks